lib/matplotlib/rcsetup.py

```diff
@@ -32,20 +33,34 @@
 from cycler import Cycler, cycler as ccycler
 
 
-# The capitalized forms are needed for ipython at present; this may
-# change for later versions.
-interactive_bk = [
-    'GTK3Agg', 'GTK3Cairo', 'GTK4Agg', 'GTK4Cairo',
-    'MacOSX',
-    'nbAgg',
-    'QtAgg', 'QtCairo', 'Qt5Agg', 'Qt5Cairo',
-    'TkAgg', 'TkCairo',
-    'WebAgg',
-    'WX', 'WXAgg', 'WXCairo',
-]
-non_interactive_bk = ['agg', 'cairo',
-                      'pdf', 'pgf', 'ps', 'svg', 'template']
-all_backends = interactive_bk + non_interactive_bk
+# Deprecation of module-level attributes using PEP 562
+_deprecated_interactive_bk = backendRegistry.list_builtin(BackendFilter.INTERACTIVE)
```

Comment:
If we didn't deprecate these 3 attributes, this block of changes would be replaced by
```python
interactive_bk = backendRegistry.list_builtin(BackendFilter.INTERACTIVE)
non_interactive_bk = backendRegistry.list_builtin(BackendFilter.NON_INTERACTIVE)
all_backends = backendRegistry.list_builtin()
```